test/tla/cli-integration-tests.md

```diff
@@ -1793,6 +1793,38 @@ Type checker [OK]
 EXITCODE: OK
 ```
 
+### typecheck bug #860
+
+Unhandled exception thrown when type-checking a spec that uses the wrong
+annotation syntax for operators.
+
+See https://github.com/informalsystems/apalache/issues/860
+
+```sh
+$ apalache-mc typecheck Bug860.tla | sed 's/[IEW]@.*//'
+...
+Parsing error in the type annotation:  (Int, Int) -> Bool
+Typing input error: Parser error in type annotation of Op: '=>' expected but -> found
+...
+EXITCODE: ERROR (255)
+```
+
+### typecheck bug #832
+
+Unhandled exception thrown due to incorrect annotation of a tuple return
+type.
+
+See https://github.com/informalsystems/apalache/issues/832
+
+```sh
+$ apalache-mc typecheck Bug832.tla | sed 's/[IEW]@.*//'
+...
+Parsing error in the type annotation:  () => (Bool, Bool)
+Typing input error: Parser error in type annotation of Example: '->' expected but ) found
```

Comment:
I wonder, if the users would be able to figure out that they should use `<<...>>` instead of `(...)` by reading this message.

Reply:
I don't think so at all!

My understanding of the current parser is that we'll need to change the structure a bit to supply more informative errors. I think we'll need a way of identify "junk" token sequences as a fallback, which we can then use to report specific errors and how to resolve them to the user. I hope to do this (or help see it done) as a followup.